src/libasr/codegen/asr_to_wasm.cpp

```diff
@@ -328,7 +328,27 @@ class ASRToWASMVisitor : public ASR::BaseVisitor<ASRToWASMVisitor> {
             if (ASR::is_a<ASR::Function_t>(*item.second)) {
                 ASR::Function_t *s =
                     ASR::down_cast<ASR::Function_t>(item.second);
-                this->visit_Function(*s);
+                bool hasTypeParameter = false;
+                for (size_t i = 0; (i < s->n_args) && (!hasTypeParameter); i++) {
+                    ASR::Variable_t *arg = ASRUtils::EXPR2VAR(s->m_args[i]);
+                    if (arg->m_type->type == ASR::ttypeType::TypeParameter) {
+                        hasTypeParameter = true;
+                    }
+                }
+
+                auto &fn_scope = s->m_symtab->get_scope();
+                for (auto itr = fn_scope.begin(); (itr != fn_scope.end()) && (!hasTypeParameter); itr++ ) {
+                    if (ASR::is_a<ASR::Variable_t>(*itr->second)) {
+                        ASR::Variable_t *v = ASR::down_cast<ASR::Variable_t>(itr->second);
+                        if (v->m_type->type == ASR::ttypeType::TypeParameter) {
+                            hasTypeParameter = true;
+                        }
+                    }
+                }
```

Comment:
I think this can be written as:
```suggestion
                bool hasTypeParameter = (s->n_type_params > 0);
```

Reply:
Done. Thank you for the guidance.